apps/test/unit/code-studio/components/header/SignInCalloutTest.js

```diff
@@ -20,3 +21,19 @@ describe('ViewPopup', () => {
     expect(wrapper.html().includes('user-not-signed-in.png')).to.be.true;
   });
 });
+
+describe('Check popup does not appear when flag is set', () => {
+  it('does not obscure the current view', () => {
+    wrapper.setState({hideCallout: true});
+    expect(wrapper.html() === null).to.be.true;
+  });
+});
+
+describe('The click to dismiss correctly updates the flag', () => {
+  it('shows prior to click, and dismisses after', () => {
+    parentWrapper.setState({hideCallout: false});
+    expect(parentWrapper.html().includes('uitest-signincallout')).to.be.true;
+    parentWrapper.find('.modal-backdrop').simulate('click');
+    expect(parentWrapper.html() === null).to.be.true;
+  });
```

Comment:
Another test you could add here is checking that if the cookie or session storage is set that the modal doesn't display with a stub like: `sinon.stub(cookies, 'get').withArgs('hide_signin_callout').returns('true')`

Reply:
Oo very cool- Will do!

Reply:
Added this to check both cookies and session storage!